libdbus-sys/build_vendored.rs

```diff
@@ -0,0 +1,370 @@
+//! Vendored Libdbus build.
+//! 
+//! Configures + Compiles + Statically Links a pinned libdbus release.
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::process::Command;
+use std::path::Path;
+use std::error::Error;
+use std::string::ToString;
+use std::collections::hash_map::Iter;
+
+// Version of vendored libdbus
+const DBUS_VERSION: &str = "\"1.15.2\"";
+const DBUS_VERSION_MAJOR: u32 = 1;
+const DBUS_VERSION_MINOR: u32 = 15;
+const DBUS_VERSION_MICRO: u32 = 2;
+
+// Location of dbus-daemon socket
+const DBUS_SYSTEM_BUS_DEFAULT_ADDRESS: &str = "/run/dbus/system_bus_socket";
+const DBUS_SESSION_SOCKET_DIR: &str = "/tmp";
+const DBUS_MACHINE_UUID_FILE: &str = "/etc/machine-id";
+
+/// Source Files Required
+const SOURCE_FILES: &'static [&'static str] = &[
```

Comment:
I was thinking, to make it as easy as possible to update this file when necessary,
Can you put references on each section? So `SOURCE_FILES` would be:
```
// From vendor/dbus/dbus/meson.build
const DBUS_LIB_SOURCES: &'static [&'static str] = &[
```

...having the same name of the constant is preferred, if this is not practical then add a comment what sections you combined. It does not have to be a step-by-step explanation or anything, just a quick one-line comment.

Reply:
Yeah that's a good idea. Will do.

Reply:
Added in https://github.com/diwic/dbus-rs/pull/408/commits/0f1afd0b962a1d9e6043c97bff98fedd26c7933f

Reply:
And variable names in https://github.com/diwic/dbus-rs/pull/408/commits/94f8e697916d397b34fc44b77f5a433265c87c0b